### ISSUE TITLE
Better structure to client side code

### ISSUE THREAD

**Issue by local-minimum** [ludumdare/ludumdare#1804]
I ought to be one of the most familiar with the client side code and I find it very confusing, hard to navigate, and easy to introduce bugs when working with.

The way I see it a lot could be helped if we:

- [ ] Remove or at least reduced globals. Functions like `cN` should be imported in the classes that use them to avoid magic features and make it easy finding these functions while working in the code.
- [ ] Split up complex classes into more sensible units. Some examples:
 * The `Node`:s and their subtypes should not mix the display and edit view in a single component, there should be a dedicated minimal class that only handles swapping between the two.
 * The same goes for `Note`:s / comments.
 * Complex blocks of the `Item` should be own components in separate files, `ItemPlatformLink`,  the stuff that marks something like a submitted/game etc, `ItemMetrcis`, `ItemVote`, `ItemGrades`. There's probably more.
- [ ] Use proper state handling. Preferrably with both Prop Types (via https://github.com/developit/preact-compat) and Redux (https://github.com/developit/preact-redux). This would help clean up immensly, would remove the use of many globals like (`node_CanGrade` and similar) and allow rendering components to be much more focused on their task. A lot of state logic is scattered an repeated all over the place.

**Comment by Noojuno** [comment]
I'm all for this and completely willing to commit time to this

**Comment by local-minimum** [comment]
Great, we just got to make sure we do it stepwise and coordinate the effort so we don't block and conflict each other

**Comment by noah-potter** [comment]
I agree with everything you said. I'd also like to suggest using prettier as a code formatter. There's a bunch of conflicting code styles, even within the same block of code, that make following the complex logic even more difficult. An easy first step would being running prettier on the entire project and moving from there.

**Comment by mikekasprzak** [comment]
I've tackled some of these issues in #1998 (cN, simplifying, VSCode/JSdoc support), intermittently marked things as deprecated, but there's a lot of inconsistencies and I've already broken/changed too much for an event just a few weeks away. 😅 

It's not Redux, but Preact 10 ships with support for [hooks](https://preactjs.com/guide/v10/hooks) and reworked [contexts](https://preactjs.com/guide/v10/context). These might do the job of avoiding passing too much as props, but it's going to require some rethinking. There are also some caveats to hooks (the way they trigger repaints) that this work-in-progress new addon is looking to address: https://github.com/preactjs/preact/pull/3474
